### PR TITLE
hardware overview: add 'Start Here' section

### PR DESCRIPTION
**Pull request by szczys**
add a matrix of hardware vendors that outlines Golioth SDK support and links to quickstart guides.

### CODE REVIEW

File: docs/hardware/1-home.md [hunk 1]
```diff
@@ -58,4 +83,4 @@ low development effort.
 
 If you're interested in using a board from this category, you can reach out to us on
 [Discord](https://golioth.io/discord), and we can help you with next steps to
```
Comment (dylanswartz) [inline docs/hardware/1-home.md:85]: Do we want to change this to the forum instead of discord? (I recall a general desire to shift folks there)

**Review by dylanswartz — APPROVED** [review]
Looks good!

File: docs/hardware/1-home.md (hunk 1 above)
Comment (szczys) [reply docs/hardware/1-home.md:85]: Out of the scope of this PR but I can add it.

**Comment by ChrisGammell** [comment]
Another one that might be out of scope, but what about this page? Also references specific hardware, might be good to just bounce them to the "Hardware Overview" page? 

https://golioth-docs-dev--pr284-szczys-hardware-add-abd2r9pj.web.app/getting-started/console/choose-device

**Pull request merged**